### PR TITLE
feat(ims/ecs_system_image): support `ecs_system_image` resource

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:

Support `ecs_system_image` resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

1. Support `ecs_system_image` resource.
2. Some methods are defined as public methods, which facilitate reuse when expanding other resources in the future.

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.

```
$ export HW_ENTERPRISE_PROJECT_ID_TEST=xxxxxx
$ make testacc TEST="./huaweicloud/services/acceptance/ims" TESTARGS="-run TestAccEcsSystemImage_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims -v -run TestAccEcsSystemImage_basic -timeout 360m -parallel 4
=== RUN   TestAccEcsSystemImage_basic
=== PAUSE TestAccEcsSystemImage_basic
=== CONT  TestAccEcsSystemImage_basic
--- PASS: TestAccEcsSystemImage_basic (603.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       603.676s

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
image

### CODE REVIEW

File: docs/resources/ims_ecs_system_image.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the image.
```
Comment (deer-hang) [inline docs/resources/ims_ecs_system_image.md:34]: Please add detailed descriptions of fields according to the document.
https://support.huaweicloud.com/intl/en-us/api-ims/ims_03_0901.html

File: docs/resources/ims_ecs_system_image.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 20 minutes.
+* `delete` - Default is 10 minutes.
+
+## Import
+
+The ecs system Image can be imported using the `id`, e.g.
```
Comment (deer-hang) [inline docs/resources/ims_ecs_system_image.md]:
```suggestion
The ECS system Image can be imported using the `id`, e.g.
```
Abbreviations with special meanings need to be capitalized.

File: docs/resources/ims_ecs_system_image.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID to which the IMS image belongs.
+  For enterprise users, if omitted, default enterprise project will be used.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the image.
+
+* `status` - The status of the image.
```
Comment (deer-hang) [inline docs/resources/ims_ecs_system_image.md:57]: Please list the enumeration value and meaning of `status`.

File: huaweicloud/services/acceptance/ims/resource_huaweicloud_ims_ecs_system_image_test.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  min_ram               = %[5]d
+  max_ram               = %[6]d
+
+  tags = {
+    foo  = "bar"
+    key  = "value1"
+    key2 = "value2"
+  }
+}
+`, testAccEcsSystemImage_base(rName), rName, rNameUpdate, migrateEpsId, minRAM, maxRAM)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ims/resource_huaweicloud_ims_ecs_system_image_test.go]:
```suggestion
`, testAccEcsSystemImage_base(rName), rNameUpdate, migrateEpsId, minRAM, maxRAM)
```
The second parameter `rName` is not used.

File: huaweicloud/services/acceptance/ims/resource_huaweicloud_ims_ecs_system_image_test.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  min_ram               = %[5]d
+  max_ram               = %[6]d
+
+  tags = {
+    foo  = "bar"
+    key  = "value1"
+    key2 = "value2"
+  }
+}
+`, testAccEcsSystemImage_base(rName), rName, rNameUpdate, defaultEpsId, minRAM, maxRAM)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ims/resource_huaweicloud_ims_ecs_system_image_test.go:208]: ditto

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+		region     = cfg.GetRegion(d)
+		createResp *cloudimages.JobResponse
+	)
+
+	client, err := cfg.ImageV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IMS v2 client: %s", err)
+	}
+
+	imageTags := BuildCreateImageTagsParam(d)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go]:
```suggestion
	imageTags := buildCreateImageTagsParam(d)
```
Please check other places where similar issues exist.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+		EnterpriseProjectID: cfg.GetEnterpriseProjectID(d),
+	}
+	createResp, err = cloudimages.CreateImageByServer(client, createOpts).ExtractJobResponse()
+	if err != nil {
+		return diag.Errorf("error creating IMS image: %s", err)
+	}
+
+	imageId, err := WaitForCreateImageCompleted(client, d, createResp.JobID)
+	if err != nil {
+		return diag.Errorf("error waiting for IMS create ecs system image to complete: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go]:
```suggestion
		return diag.Errorf("error waiting for IMS create ECS system image to complete: %s", err)
```
Please check other places where similar issues exist.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+		Description:         d.Get("description").(string),
+		InstanceId:          d.Get("instance_id").(string),
+		MaxRam:              d.Get("max_ram").(int),
+		MinRam:              d.Get("min_ram").(int),
+		ImageTags:           imageTags,
+		EnterpriseProjectID: cfg.GetEnterpriseProjectID(d),
+	}
+	createResp, err = cloudimages.CreateImageByServer(client, createOpts).ExtractJobResponse()
+	if err != nil {
+		return diag.Errorf("error creating IMS image: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go]: It is best to separate the logs of different paths to create images.
```suggestion
		return diag.Errorf("error creating ECS system IMS image: %s", err)
```

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 9]
```diff
@@ -0,0 +1,582 @@
...
+		return "", err
+	}
+
+	imageId, err := cloudimages.GetJobEntity(client, jobId, "image_id")
+	if err != nil {
+		return "", err
+	}
+
+	if _, ok := imageId.(string); !ok {
+		return "", fmt.Errorf("an unexpected conversion error occurred with image_id")
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go:193]:
```suggestion
		return "", fmt.Errorf("an unexpected conversion error occurred with image_id: %s", err)
```

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 10]
```diff
@@ -0,0 +1,582 @@
...
+		d.Set("disk_format", image.DiskFormat),
+		d.Set("data_origin", image.DataOrigin),
+		d.Set("os_version", image.OsVersion),
+		d.Set("active_at", image.ActiveAt),
+		d.Set("created_at", image.CreatedAt.Format(time.RFC3339)),
+		d.Set("updated_at", image.UpdatedAt.Format(time.RFC3339)),
+	)
+
+	// The `max_ram` field, API returns a string type, so it needs to be converted to an int type.
+	if maxRAM, err := strconv.Atoi(image.MaxRam); err == nil {
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go]: Extract into a method and leave a log error message.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go [hunk 11]
```diff
@@ -0,0 +1,582 @@
...
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			listResp, err := GetImageList(client, d.Id())
+			if err != nil {
+				return nil, "ERROR", nil
+			}
+
+			if len(listResp) < 1 {
+				return listResp, "COMPLETED", nil
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go]:
```suggestion
				return "success", "COMPLETED", nil
```

**Comment by jinyangyang222** [comment]
make testacc TEST="./huaweicloud/services/acceptance/ims" TESTARGS="-run TestAccEcsSystemImage_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims -v -run TestAccEcsSystemImage_basic -timeout 360m -parallel 4
=== RUN   TestAccEcsSystemImage_basic
=== PAUSE TestAccEcsSystemImage_basic
=== CONT  TestAccEcsSystemImage_basic
--- PASS: TestAccEcsSystemImage_basic (512.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       512.807s

File: docs/resources/ims_ecs_system_image.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the image.
+  The valid length is limited from `1` to `128` characters.
+  The first and last letters of the name cannot be spaces.
+  The name can contain uppercase letters, lowercase letters, numbers, spaces, chinese, and special characters (-._).
+
+* `instance_id` - (Required, String, ForceNew) Specifies the source ECS instance ID used to create the image.
+
```
Comment (deer-hang) [inline docs/resources/ims_ecs_system_image.md:41]: Changing this will XXX.

File: docs/resources/ims_ecs_system_image.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID to which the IMS image belongs.
+  For enterprise users, if omitted, default enterprise project will be used.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the image.
+
+* `status` - The status of the image.
```
Comment (deer-hang) [inline docs/resources/ims_ecs_system_image.md]: Please list the enumeration values ​​for this field.

**Pull request merged**